// homey-oauth2app-tuya/lib/TuyaOAuth2App.js
'use strict';

const { OAuth2App } = require('homey-oauth2app');
const TuyaOAuth2Client = require('./TuyaOAuth2Client');

/**
 * @extends Homey.App
 * @hideconstructor
 */
class TuyaOAuth2App extends OAuth2App {

  static OAUTH2_CLIENT = TuyaOAuth2Client;
  static OAUTH2_TUYA_PROXY_URL = 'https://{{brand}}.tuya.athom.com/brand/{{brand}}';

  /**
   * Set this to your Tuya Project ID, as provided to Athom.
   * @type string
   */
  static OAUTH2_TUYA_PROXY_BRAND = null;

  async onInit(...props) {
    await super.onInit(...props);

    // Brand
    const brand = this.constructor.OAUTH2_TUYA_PROXY_BRAND;
    if (!brand) {
      throw new Error('Missing App.OAUTH2_TUYA_PROXY_BRAND');
    }

    this.proxyUrl = this.constructor.OAUTH2_TUYA_PROXY_URL
      .replace(/\{\{brand\}\}/g, brand);

    this.log(`Proxy URL: ${this.proxyUrl}`);

    this.setOAuth2Config({
      clientId: 'dummy',
      clientSecret: 'dummy',
      apiUrl: `${this.proxyUrl}/api`,
      tokenUrl: `${this.proxyUrl}/api/v1.0/token`,
      authorizationUrl: `${this.proxyUrl}/authorise`,
    });
  }

}

module.exports = TuyaOAuth2App;
